tools/SetupFlow/DevHome.SetupFlow/Services/WinGetFeaturedApplicationsDataSource.cs

```diff
@@ -0,0 +1,132 @@
+﻿﻿// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using DevHome.Common.Extensions;
+using DevHome.Common.Services;
+using DevHome.SetupFlow.Common.Helpers;
+using DevHome.SetupFlow.Models;
+using Microsoft.Windows.DevHome.SDK;
+
+namespace DevHome.SetupFlow.Services;
+
+/// <summary>
+/// Class to get featured applications from extension providers.
+/// </summary>
+public class WinGetFeaturedApplicationsDataSource : WinGetPackageDataSource
+{
+    private readonly IPluginService _extensionService;
+    private readonly IList<IFeaturedApplicationsGroup> _groups;
+
+    public WinGetFeaturedApplicationsDataSource(IWindowsPackageManager wpm, IPluginService extensionService)
+        : base(wpm)
+    {
+        _extensionService = extensionService;
+        _groups = new List<IFeaturedApplicationsGroup>();
+    }
+
+    public override int CatalogCount => _groups.Count;
+
+    public async override Task InitializeAsync()
+    {
+        var extensions = await _extensionService.GetInstalledPluginsAsync(ProviderType.FeaturedApplications);
+        Log.Logger?.ReportInfo(Log.Component.AppManagement, "Initializing featured applications from all extensions");
+        foreach (var extension in extensions)
+        {
+            var extensionName = extension.Name;
+            try
+            {
+                Log.Logger?.ReportInfo(Log.Component.AppManagement, $"Getting featured applications provider from extension '{extensionName}'");
+                var provider = await extension.GetProviderAsync<IFeaturedApplicationsProvider>();
+                if (provider != null)
+                {
+                    var groupsResult = await provider.GetFeaturedApplicationsGroupsAsync();
+                    if (groupsResult.Result.Status == ProviderOperationStatus.Success)
+                    {
+                        var groups = groupsResult.FeaturedApplicationsGroups;
+                        Log.Logger?.ReportInfo(Log.Component.AppManagement, $"Found {groups.Count} groups from extension '{extensionName}'");
+                        for (var i = 0; i < groups.Count; ++i)
```

Comment:
No issue but I can hear Flor advising us to use linq here

Reply:
Oops, I forgot to add a comment here. We cannot use linq or foreach on out of proc vectors:
 https://github.com/microsoft/CsWinRT/issues/1205